pkg/karmadactl/cmdinit/kubernetes/services_test.go

```diff
@@ -0,0 +1,109 @@
+package kubernetes
+
+import (
+	"context"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestCommandInitOption_makeEtcdService(t *testing.T) {
+	cmdOpt := CommandInitOption{EtcdReplicas: 1, Namespace: "karmada"}
+	svc := cmdOpt.makeEtcdService("foo")
+	if svc == nil {
+		t.Errorf("makeEtcdService() return nil")
+	}
+}
+
+func TestCommandInitOption_makeKarmadaAPIServerService(t *testing.T) {
+	cmdOpt := CommandInitOption{EtcdReplicas: 1, Namespace: "karmada"}
+	svc := cmdOpt.makeKarmadaAPIServerService()
+	if svc == nil {
+		t.Errorf("makeKarmadaAPIServerService() return nil")
+	}
+}
+
+func TestCommandInitOption_kubeControllerManagerService(t *testing.T) {
+	cmdOpt := CommandInitOption{EtcdReplicas: 1, Namespace: "karmada"}
+	svc := cmdOpt.kubeControllerManagerService()
+	if svc == nil {
+		t.Errorf("kubeControllerManagerService() return nil")
+	}
+}
+
+func TestCommandInitOption_karmadaWebhookService(t *testing.T) {
+	cmdOpt := CommandInitOption{EtcdReplicas: 1, Namespace: "karmada"}
+	svc := cmdOpt.karmadaWebhookService()
+	if svc == nil {
+		t.Errorf("karmadaWebhookService() return nil")
+	}
+}
+
+func TestCommandInitOption_karmadaAggregatedAPIServerService(t *testing.T) {
+	cmdOpt := CommandInitOption{EtcdReplicas: 1, Namespace: "karmada"}
+	svc := cmdOpt.karmadaAggregatedAPIServerService()
+	if svc == nil {
+		t.Errorf("karmadaAggregatedAPIServerService() return nil")
+	}
+}
+
+func TestCommandInitOption_isNodePortExist(t *testing.T) {
+	tests := []struct {
+		name   string
+		option CommandInitOption
+		svc    []*corev1.Service
+		exists bool
+	}{
+		{
+			name: "there is no svc",
+			option: CommandInitOption{
+				KubeClientSet:            fake.NewSimpleClientset(),
+				KarmadaAPIServerNodePort: 30000,
+			},
+			exists: false,
+		},
+		{
+			name: "there is no node port svc",
+			option: CommandInitOption{
+				KubeClientSet:            fake.NewSimpleClientset(),
+				KarmadaAPIServerNodePort: 30000,
+			},
+			svc: []*corev1.Service{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+					Spec:       corev1.ServiceSpec{Type: corev1.ServiceTypeClusterIP, Ports: []corev1.ServicePort{{Port: 30000}}},
+				},
+			},
+			exists: false,
+		},
+		{
+			name: "there is node port svc",
```

Comment:
How about adding one more case: `exist a NodePort type service with the same nodePort`